File: nexo/lib/shared/models/address.dart
class Address {
  final int id;
  final String label;
  final String recipientName;
  final String phone;
  final String street;
  final String exteriorNumber;
  final String interiorNumber;
  final String neighborhood;
  final String city;
  final String state;
  final String postalCode;
  final String references;
  final double? latitude;
  final double? longitude;
  final bool isDefault;
  final DateTime? createdAt;
  final String fullAddress;

  const Address({
    required this.id,
    required this.label,
    required this.recipientName,
    required this.phone,
    required this.street,
    required this.exteriorNumber,
    required this.interiorNumber,
    required this.neighborhood,
    required this.city,
    required this.state,
    required this.postalCode,
    required this.references,
    required this.latitude,
    required this.longitude,
    required this.isDefault,
    required this.createdAt,
    required this.fullAddress,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: (json['id'] ?? 0) as int,
      label: (json['label'] ?? '') as String,
      recipientName: (json['recipientName'] ?? '') as String,
      phone: (json['phone'] ?? '') as String,
      street: (json['street'] ?? '') as String,
      exteriorNumber: (json['exteriorNumber'] ?? '') as String,
      interiorNumber: (json['interiorNumber'] ?? '') as String,
      neighborhood: (json['neighborhood'] ?? '') as String,
      city: (json['city'] ?? '') as String,
      state: (json['state'] ?? '') as String,
      postalCode: (json['postalCode'] ?? '') as String,
      references: (json['references'] ?? '') as String,
      latitude: json['latitude'] == null
          ? null
          : (json['latitude'] as num).toDouble(),
      longitude: json['longitude'] == null
          ? null
          : (json['longitude'] as num).toDouble(),
      isDefault: (json['isDefault'] ?? false) as bool,
      createdAt: json['createdAt'] == null
          ? null
          : DateTime.tryParse(json['createdAt'].toString()),
      fullAddress: (json['fullAddress'] ?? '') as String,
    );
  }
}

File: api nexo/Nexo/Nexo.Api/Entitites/Address.cs
namespace Nexo.Api.Entitites
{
    public class Address
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public string Label { get; set; } = string.Empty;
        public string RecipientName { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Street { get; set; } = string.Empty;
        public string ExteriorNumber { get; set; } = string.Empty;
        public string InteriorNumber { get; set; } = string.Empty;
        public string Neighborhood { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string State { get; set; } = string.Empty;
        public string PostalCode { get; set; } = string.Empty;
        public string References { get; set; } = string.Empty;
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public bool IsDefault { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
